examples_tests/__main__.py

```diff
@@ -14,48 +14,47 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import argparse
+"""Snapcraft examples tests.
+
+Usage:
+  examples_tests [--skip-install] [--ip IP_OR_HOSTNAME]
+                 [--port PORT_NUMBER] [--filter REGEXP]
+                 [--subunit]
+
+Options:
+  --skip-install       skip the tests that install the example snaps into a
+                       snapp test bed.
+  --ip IP_OR_HOSTNAME  IP of the testbed. If no IP is passed, a virtual
+                       machine will be created for the test.
+  --port PORT_NUMBER   SSH port of the testbed. Defaults to use port 22.
+  --filter REGEXP      a regular expression to filter the examples to test.
+  --subunit            generate subunit results.
+
+"""
+
 import logging
 import sys
 
+import docopt
+
 from examples_tests import tests
 
 
 def main():
     logging.basicConfig(level=logging.INFO)
 
-    parser = argparse.ArgumentParser()
-    parser.add_argument(
-        '--skip-install',
-        help=('skip the tests that install the example snaps into a snappy '
-              'virtual machine.'),
-        action='store_true')
-    parser.add_argument(
-        '--ip',
-        help=('IP of the testbed. If no IP is passed, a virtual machine will '
-              'be created for the test.'))
-    parser.add_argument(
-        '--port',
-        help=('SSH port of the testbed. Defaults to use port 22.'))
-    parser.add_argument(
-        '--filter',
-        help=('a regular expression to filter the examples to test.'))
-    parser.add_argument(
-        '--subunit',
-        help='generate subunit results.',
-        action='store_true')
+    arguments = docopt.docopt(__doc__)
 
-    args = parser.parse_args()
-    if args.skip_install:
+    if arguments['--skip-install']:
         tests.config['skip-install'] = True
-    if args.ip:
-        tests.config['ip'] = args.ip
-    if args.port:
-        tests.config['port'] = args.port
-    if args.filter:
-        tests.config['filter'] = args.filter
+    if arguments['--ip']:
+        tests.config['ip'] = arguments['--ip']
+    if arguments['--port']:
+        tests.config['port'] = arguments['--port']
+    if arguments['--filter']:
+        tests.config['filter'] = arguments['--filter']
```

Comment:
Only the difference between having a dictionary element with None as a value and not having it.
>>> {}.get('port', 22)
22
>>> {'port': None}.get('port', 22)
>>>

I'll make the change to see if you like it better. I slightly prefer to put the complexity here than on the test; but on the other hand it sounds better to handle '', None and unexisting.